src/main/java/messenger/MessengerDatabase.java

```diff
@@ -0,0 +1,43 @@
+package messenger;
+
+import notification.Notification;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.util.HashMap;
+import java.util.HashSet;
+
+public class MessengerDatabase {
+
+    private HashMap<String, String> idToNameMap;
+
+    private HashSet<Notification> storedNotifications;
+
+    public MessengerDatabase() {
+        storedNotifications = new HashSet<>();
+        try {
```

Comment:
Doesn't new `try` handle this streams better? Consider this -
```try (final FileInputStream fis =
                 new FileInputStream("src/main/resources/idToName.ser");
            final ObjectInputStream ois =
                 new ObjectInputStream(fis)) {
            idToNameMap = (HashMap) ois.readObject();
        } catch (IOException ioe) {
            ioe.printStackTrace();
        } catch (ClassNotFoundException c) {
            System.out.println("Class not found");
            c.printStackTrace(); 
      }
```
However, if you want to stick with your `try` you need to close streams in `finally` block to make suer the will be closed no matter what

Reply:
I don't understand that :disappointed:

Reply:
There is `try-with-resources` in Java. It handles the resource you are using in try block
https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Wait, my code is pasted in half, lemme fix that